Lib/aifc.py

```diff
@@ -543,16 +549,19 @@ class Aifc_write:
 
     def __init__(self, f):
         if isinstance(f, str):
-            filename = f
-            f = builtins.open(f, 'wb')
-        else:
-            # else, assume it is an open file object already
-            filename = '???'
-        self.initfp(f)
-        if filename[-5:] == '.aiff':
-            self._aifc = 0
+            file_object = builtins.open(f, 'wb')
+            try:
+                self.initfp(file_object)
+            except:
+                file_object.close()
+                raise
+
+            # treat .aiff file extensions as non-compressed audio
+            if f.endswith('.aiff'):
```

Comment:
Ah, sorry, I missed this. It would be better to write just `self._aifc = not f.endswith('.aiff')` if you want to modernize this code.

But now it looks to me that `initfp()` never fails. All changes of `Aifc_write` were not needed.